terraform/charm/main.tf

```diff
@@ -0,0 +1,62 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+terraform {
+  required_providers {
+    juju = {
+      source  = "juju/juju"
+      version = ">= 0.19.0"
+    }
+  }
+}
+
+resource "juju_application" "haproxy" {
+  name  = var.app_name
+  model = var.model
+  units = var.units
+
+  charm {
+    name     = "haproxy"
+    revision = var.revision
+    channel  = var.channel
+    base     = var.base
+  }
+
+  config = var.config
+
+  expose {}
+}
+
+resource "juju_application" "hacluster" {
+  count = var.use_hacluster ? 1 : 0
+  name  = "hacluster"
```

Comment:
We should probably allow to change this `name`

Reply:
makes sense, I updated this